lib/core/src/send_swap.rs

```diff
@@ -93,14 +94,24 @@ impl SendSwapHandler {
                 Ok(())
             }
 
-            // Boltz has detected the lockup in the mempool, we can speed up
-            // the claim by doing so cooperatively
+            // Boltz has detected the lockup in the mempool. If the swap is not to be batched
+            // we can speed up the claim by doing so cooperatively.
             SubSwapStates::TransactionClaimPending => {
                 if swap.metadata.is_local {
-                    self.cooperate_claim(&swap).await.map_err(|e| {
-                        error!("Could not cooperate Send Swap {id} claim: {e}");
-                        anyhow!("Could not post claim details. Err: {e:?}")
-                    })?;
+                    let pair = swap.get_boltz_pair()?;
+                    if swap.receiver_amount_sat < pair.limits.minimal {
+                        // The swap is to be batched and does not need to be cooperatively claimed
+                        self.get_claim_details_and_validate_preimage(&swap)
```

Comment:
This one (although has a "get" notation) updates the swap state which might make it a bit difficult to follow. 
How about calling this method anyway (also on the cooperative claim), updating the swap state (using the result) and also pass the result to cooperative_claim function?